### ISSUE TITLE
Cluster Validation should reattempt after flapping healthiness

### ISSUE THREAD

**Issue by rifelpet** [kubernetes/kops#11920]
/kind bug

In this prow job:

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/e2e-kops-grid-u2004-k20-ko20-containerd/1409324603305627648

With `kops validate cluster --count 10 --wait 15m0s` we observe:

```
 W0628 01:42:03.506979    4115 validate_cluster.go:221] (will retry): cluster not yet healthy 
 W0628 01:42:16.499706    4115 validate_cluster.go:221] (will retry): cluster not yet healthy 
 W0628 01:42:29.534311    4115 validate_cluster.go:221] (will retry): cluster not yet healthy 
 I0628 01:42:42.527542    4115 validate_cluster.go:209] (will retry): cluster passed validation 1 consecutive times 
```
followed by:

```
 VALIDATION ERRORS
KIND	NAME									MESSAGE
Pod	kube-system/kube-proxy-ip-172-20-50-93.ap-south-1.compute.internal	system-node-critical pod "kube-proxy-ip-172-20-50-93.ap-south-1.compute.internal" is pending
Validation Failed
Validation failed: cluster not yet healthy 
```

which causes the `validate cluster` to exit non-zero after a total of ~5 minutes. I would expect any flapping in healthiness to be acceptable by `validate cluster` as long as 10 consecutive validations pass before the 15 minute timeout. In this case kops should just reset the consecutive count back to 0 and keep reattempting validation.

**Comment by hakman** [comment]
@johngmyers I think you wanted implemented this way to catch flakes during testing, do you remember?

**Comment by johngmyers** [comment]
That behavior isn't making sense. By my reading of the code, in order for the `Validation failed: cluster not yet healthy` message to be logged, `options.wait` would have to be written over with a nonpositive value. But nothing in the code besides pflags writes to `options.wait`. So perhaps something in cluster validation is writing out of bounds?

**Comment by hakman** [comment]
@rifelpet I think the issue comes form the kOps version this test is using: https://storage.googleapis.com/kops-ci/markers/release-1.20/latest-ci-updown-green.txt.
You already fixed the issue in master with https://github.com/kubernetes/kops/pull/11049, but it is only available in kOps 1.21.
